### PR TITLE
fix: Fixed parameter naming for compatibility with ROOT import/export

### LINKED ISSUE #1107: fixed information faulty in validation example
# Description

```
pyhf xml2json validation/xmlimport_input/config/example.xml  --basedir validation/xmlimport_input
```

produces

```
                    {
                        "fixed": true,
                        "name": "alpha_syst1"
                    }
```

whereas it should produce (I think)

```
                    {
                        "fixed": true,
                        "name": "syst1"
                    }
```

### PR DESCRIPTION
**Pull request by kratsg**
# Description

Resolves #1107. ROOT, in all its glory, decides to prefix parameter names in the measurement specification. We need to update `readxml.py` to strip this prefix, but additionally update `writexml.py` to prepend this prefix.

# Checklist Before Requesting Reviewer

- [x] Tests are passing
- [x] "WIP" removed from the title of the pull request
- [x] Selected an Assignee for the PR to be responsible for the log summary

# Before Merging

For the PR Assignees:

- [x] Summarize commit messages into a comprehensive review of the PR
```
* Fix parameter naming for compatibility with ROOT import/export
  - prefixes alpha_ and gamma_ are stripped when importing from ROOT workspaces
  - prefixes alpha_ and gamma_ are added when exporting to ROOT workspaces
```

### CODE REVIEW

File: src/pyhf/writexml.py [hunk 1]
```diff
@@ -58,7 +60,15 @@ def indent(elem, level=0):
             elem.tail = i
 
 
-def build_measurement(measurementspec):
+def build_measurement(measurementspec, modifiertypes):
+    # need to determine prefixes
```
Comment (matthewfeickert) [inline src/pyhf/writexml.py:64]: As this in in the public API `pyhf.writexml.build_measurement` can we get a docstring?

**Review by matthewfeickert — COMMENTED** [review]
One request but other than that looks good @kratsg

**Review by matthewfeickert — APPROVED** [review]
Thanks @kratsg

File: src/pyhf/readxml.py [hunk 2]
```diff
@@ -258,6 +258,8 @@ def process_measurements(toplvl, other_parameter_configs=None):
             # might be specifying multiple parameters in the same ParamSetting
             if param.text:
                 for param_name in param.text.split(' '):
+                    param_name = utils.remove_prefix(param_name, 'alpha_')
+                    param_name = utils.remove_prefix(param_name, 'gamma_')
```
[the remark this answers is not in the document]
Comment (kratsg) [reply src/pyhf/readxml.py:280]: raised ValueError.

**Review by lukasheinrich — APPROVED** [review]

**Pull request merged**